### ISSUE TITLE
Invalid value for '--parameter-overrides':   is not in valid format.

### ISSUE THREAD

**Issue by k-ogawa-1988** [CircleCI-Public/aws-sam-serverless-orb#50]
## Orb version:

4.0.0

## What happened:

When specifying `parameter-overrides` and not specifying `additional-args`, CLI will happen error.

Error message:
```
+ sam deploy --stack-name example-stack --region us-east-1 --s3-bucket example-bucket --profile default --parameter-overrides ParameterKey=dummy,ParameterValue=dummy ''
Usage: sam deploy [OPTIONS]
Try 'sam deploy -h' for help.

Error: Invalid value for '--parameter-overrides':   is not in valid format. It must look something like 'ParameterKey=KeyPairName,ParameterValue=MyKey ParameterKey=InstanceType,ParameterValue=t1.micro' or 'KeyPairName=MyKey InstanceType=t1.micro'
```

Reproduced workflow:

```yaml
version: 2.1

orbs:
  aws-cli: circleci/aws-cli@3.1.3
  aws-sam: circleci/aws-sam-serverless@4.0.0

jobs:
  lambda-edge:
    docker:
      - image: cimg/node:16.18
    steps:
      - aws-cli/setup
      - aws-sam/install
      - checkout
      - aws-sam/deploy:
          parameter-overrides: ParameterKey=dummy,ParameterValue=dummy
          s3-bucket: example-bucket
          stack-name: example-stack

workflows:
  version: 2
  sample:
    jobs:
      - lambda-edge
```

## Expected behavior:

`sam` command works fine

## Additional Information:

Just my guess, but setting `$SAM_PARAM_ADDITIONAL_ARGS` may cause this error.
If I specify `additional-args` parameters, the command line works fine.

```shell
if [ -n "$SAM_PARAM_PARAMETER_OVERRIDES" ]; then
    set -- "$@" --parameter-overrides "$(eval "echo $SAM_PARAM_PARAMETER_OVERRIDES")"
fi
if [ -z "$SAM_PARAM_ADDITIONAL_ARGS" ]; then
    set -- "$@" "$SAM_PARAM_ADDITIONAL_ARGS"  # <--- This may add '' to $@
fi

set -x
sam deploy "$@"
set +x
```

**Comment by rsorelli-hedgepoint** [comment]
Same issue here

**Error message:**
`Error: Invalid value for '--parameter-overrides':   is not in valid format. It must look something like 'ParameterKey=KeyPairName,ParameterValue=MyKey ParameterKey=InstanceType,ParameterValue=t1.micro' or 'KeyPairName=MyKey InstanceType=t1.micro'`

**Config File:**
```
version: 2.1

orbs:
  aws-cli: circleci/aws-cli@3.1.4
  aws-sam-serverless: circleci/aws-sam-serverless@4.0.0
  docker: circleci/docker@2.2.0

jobs:
  sam-build:
    docker:
      - image: 'cimg/node:12.22'
    #resource_class: xlarge
    steps:
      - checkout

      - aws-cli/setup:
          aws-access-key-id: AWS_ACCESS_KEY
          aws-region: AWS_REGION
          aws-secret-access-key: AWS_SECRET_KEY
      - aws-sam-serverless/install
      - run: sam build --parallel
      - aws-sam-serverless/deploy:
          aws-region: AWS_REGION
          capabilities: CAPABILITY_IAM,CAPABILITY_NAMED_IAM,CAPABILITY_AUTO_EXPAND
          s3-bucket: aws-sam-test
          stack-name: my-stack
          parameter-overrides: 'ParameterKey=environment,ParameterValue=prd'

workflows:
  ci-cd:
    jobs:
      - sam-build:
          context: AWS_HOM
```

**Comment by andy2mrqz** [comment]
I ran into the same thing today.  As @k-ogawa-1988 pointed out, the way the `SAM_PARAM_ADDITIONAL_ARGS` code is written seems to be the reason for the extra '' at the end of the parameter-overrides.

I found a workaround until this gets patched:

* Set `additional-args` for `sam/deploy` or `aws-sam-serverless/deploy`.  It doesn't get added to the final command, but seems to circumvent this test `[ -z "$SAM_PARAM_ADDITIONAL_ARGS" ]`
* With that out of the way, the parameter-overrides get interpreted correctly!

If I find some time I may look into creating a PR, but it would be best if the maintainers could fix this bug. (Also thanks again @k-ogawa-1988 for cluing us into that section of code as causing the issue)
